test/models/stage_test.rb

```diff
@@ -435,4 +435,38 @@
       stage.command_updated_at.to_i.must_equal t.to_i
     end
   end
+
+  describe "versioning" do
+    around { |t| PaperTrail.with_logging(&t) }
+
+    it "tracks important changes" do
+      stage.update_attribute(:name, "Foo")
+      stage.versions.size.must_equal 1
+    end
+
+    it "ignores unimportant changes" do
+      stage.update_attributes(order: 5, updated_at: 1.second.from_now)
+      stage.versions.size.must_equal 0
+    end
+
+    it "records script" do
+      stage.record_script_change
+      YAML.load(stage.versions.first.object)['script'].must_equal stage.script
+    end
+
+    it "can restore ... but loses script" do
```

Comment:
restoring script is tricky since you'd have to split it into commands and create new commands where necessary and set correct ordering ... nothing impossible, but not something I wanted to tackle :)